styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -208,14 +218,26 @@ private WorkflowConfiguration readFromJsonWithDefaults(ByteString payload)
     return WorkflowConfigurationBuilder.from(workflowConfig).deploymentTime(time.get()).build();
   }
 
-  private Response<Collection<Workflow>> workflows() {
+  private Response<Collection<Workflow>> workflows(Request request) {
     try {
-      return Response.forPayload(storage.workflows().values());
+      Map<QueryParams, String> paramFilters =
+          Map.of(DEPLOYMENT_TYPE, getFilterParams(request, DEPLOYMENT_TYPE),
+              DEPLOYMENT_TIME_BEFORE, getFilterParams(request, DEPLOYMENT_TIME_BEFORE),
+              DEPLOYMENT_TIME_AFTER, getFilterParams(request, DEPLOYMENT_TIME_AFTER));
```

Comment:
```java
var paramFilters = Stream.of(QueryParams.values())
  .map(e -> getFilterParams(request, e).map(param -> Map.entry(e, param))
  .map(Optional::stream)
  .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue);
```
This version is a bit better because if no query params are passed, then the map would be empty instead of containing params pointing to empty strings